lib/vtls/openssl.c

```diff
@@ -1878,15 +1878,45 @@ static void ossl_close(struct Curl_cfilter *cf, struct Curl_easy *data)
 
   if(backend->handle) {
     if(cf->next && cf->next->connected) {
-      char buf[32];
+      char buf[1024];
+      int nread, err;
+      long sslerr;
+
       /* Maybe the server has already sent a close notify alert.
          Read it to avoid an RST on the TCP connection. */
       (void)SSL_read(backend->handle, buf, (int)sizeof(buf));
-
-      (void)SSL_shutdown(backend->handle);
-
       ERR_clear_error();
+      if(SSL_shutdown(backend->handle)) {
```

Comment:
```suggestion
      if(SSL_shutdown(backend->handle) == 1) {
```

The manpage for [SSL_shutdown](https://www.openssl.org/docs/man3.1/man3/SSL_shutdown.html) says that 1 means finished.

Reply:
And that `0`  means "not finished". Hmm, tricky API as usual in OpenSSL.

When a function is documented to have such a BOOLEAN return value, I prefer testing it like I did.

Reply:
That's the mistake though it's not bool it's int and when it returns -1 the shutdown is not finished

Reply:
Ah, reading the man page further reveals this.